### PR TITLE
Migrate public/main to Typescript

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*
#7 
*Description of changes:*
Migrate all of `public/main` into Typescript. This includes the homepage view of the Reporting plugin page. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/components/main/reports_table.tsx [hunk 1]
```diff
@@ -34,7 +35,7 @@ export const reports_list_columns = [
   {
     field: 'reportSource',
     name: 'Source',
-    render: username => (
+    render: (username: any) => (
```
Comment (mihirsoni) [inline public/components/main/reports_table.tsx:38]: Shouldn't this be string ?

File: public/components/main/reports_table.tsx [hunk 2]
```diff
@@ -106,21 +103,27 @@ const renderToolsLeft = () => {
...
+};
+
 export const reports_list_selection_value = {
-  selectable: user => user.online,
-  selectableMessage: selectable =>
+  selectable: (user: { online: any; }) => user.online,
+  selectableMessage: (selectable: any) =>
       !selectable ? 'User is currently offline' : undefined,
-  onSelectionChange: selection => setSelection(selection),
+  onSelectionChange: (selection: any) => onSelectionChange,
```
Comment (seraphjiang) [inline public/components/main/reports_table.tsx:128]: try use specific type
```
seleactable:boolean
```

File: public/components/main/reports_table.tsx (hunk 1 above)
Comment (davidcui-amzn) [reply public/components/main/reports_table.tsx:38]: Yes, will change

File: public/components/main/reports_table.tsx (hunk 2 above)
Comment (davidcui-amzn) [reply public/components/main/reports_table.tsx:128]: Will change

File: public/components/main/scheduled_reports_table.tsx [hunk 3]
```diff
@@ -33,7 +34,7 @@ export const scheduled_report_columns = [
   {
     field: 'reportSource',
     name: 'Source',
-    render: username => (
+    render: (username: any) => (
```
Comment (zhongnansu) [inline public/components/main/scheduled_reports_table.tsx:37]: should this be `string`?

Comment (davidcui-amzn) [reply public/components/main/scheduled_reports_table.tsx:37]: good catch, will change

**Pull request merged**